lib/client/ipam_test.go

```diff
@@ -170,8 +189,73 @@ var _ = Describe("IPAM tests", func() {
+			Expect(len(outFailed)).To(Equal(args.expFailedIPs))
+
+			// Assert if an error was expected.
+			if args.expError != nil {
+				Expect(outError).To(HaveOccurred())
+				Expect(outError.Error()).To(Equal(args.expError.Error()))
+			}
+		},
+
+		// Test cases (ClaimAffinity):
+		// Test 1: claim affinity for an unclaimed IPNet of size 64 - expect 1 claimed blocks, 0 failed and expect no error.
+		Entry("Claim affinity for an unclaimed IPNet of size 64", testArgsClaimAff{"192.168.1.0/26", "host-A", true, []string{"192.168.1.0/24", "fd80:24e2:f998:72d6::/120"}, net.IP{}, 1, 0, nil}),
+
+		// Test 2: claim affinity for an unclaimed IPNet of size smaller than 64 - expect 0 claimed blocks, 0 failed and expect an error error.
+		Entry("Claim affinity for an unclaimed IPNet of size smaller than 64", testArgsClaimAff{"192.168.1.0/27", "host-A", true, []string{"192.168.1.0/24", "fd80:24e2:f998:72d6::/120"}, net.IP{}, 0, 0, errors.New("The requested CIDR (192.168.1.0/27) is smaller than the minimum.")}),
+
+		// Test 3: claim affinity for a IPNet that has an IP already assigned to another host.
+		// - Assign an IP with AssignIP to "Host-A" from a configured pool
+		// - Claim affinity for "Host-B" to the block that IP belongs to - expect 3 claimed blocks and 1 failed.
+		Entry("Claim affinity for a IPNet that has an IP already assigned to another host (Claim affinity for Host-B)", testArgsClaimAff{"10.0.0.0/24", "host-B", true, []string{"10.0.0.0/24", "fd80:24e2:f998:72d6::/120"}, net.ParseIP("10.0.0.1"), 3, 1, nil}),
+
+		// Test 4: claim affinity to a block twice from different hosts.
+		// - Claim affinity to an unclaimed block for "Host-A" - expect 4 claimed blocks, 0 failed and expect no error.
+		Entry("Claim affinity to an unclaimed block for Host-A", testArgsClaimAff{"10.0.0.0/24", "host-A", true, []string{"10.0.0.0/24", "fd80:24e2:f998:72d6::/120"}, net.IP{}, 4, 0, nil}),
+
+		// - Claim affinity to the same block again but for "host-B" this time - expect 0 claimed blocks, 4 failed and expect no error.
+		Entry("Claim affinity to the same block again but for Host-B this time", testArgsClaimAff{"10.0.0.0/24", "host-B", false, []string{"10.0.0.0/24", "fd80:24e2:f998:72d6::/120"}, net.IP{}, 0, 4, nil}),
+	)
 })
 
+// testIPAMClaimAffinity takes inNet which is the CIDR to claim affinity to. It also takes host, poolSubnets to configure a new pool,
+// when assignIP is not empty it will assign that IP and return zero-value for the return values, and cleanEnv is to clear the datastore etc.
+func testIPAMClaimAffinity(inNet net.IPNet, host string, poolSubnet []string, assignIP net.IP, cleanEnv bool) (claimed []cnet.IPNet, failed []cnet.IPNet, outErr error) {
+	if cleanEnv {
+		testutils.CleanEtcd()
+		c, _ := testutils.NewClient("")
+		for _, v := range poolSubnet {
+			testutils.CreateNewPool(*c, v, false, false, true)
+		}
+	}
+	ic := setupIPAMClient(cleanEnv)
+
+	assignIPutil(ic, assignIP, "Host-A")
```

Comment:
Why are you splitting into a `testIPAMClaimAffinity` - can't you just put all this code into the main table test function? 

It feels like not that much occurs here that couldn't just be in that function - will make this a lot easier to follow.